hive-llap/llap.sh

```diff
@@ -0,0 +1,497 @@
+readonly NODE_MANAGER_MEMORY=$(bdconfig get_property_value --configuration_file='/etc/hadoop/conf/yarn-site.xml' --name yarn.nodemanager.resource.memory-mb)
+readonly YARN_MAX_CONTAINER_MEMORY=$(bdconfig get_property_value --configuration_file='/etc/hadoop/conf/yarn-site.xml' --name yarn.scheduler.maximum-allocation-mb)
+readonly ADDITIONAL_MASTER=$(/usr/share/google/get_metadata_value attributes/dataproc-master-additional)
+readonly HAS_SSD=$(/usr/share/google/get_metadata_value attributes/ssd)
+readonly NUM_LLAP_NODES=$(/usr/share/google/get_metadata_value attributes/num-llap-nodes)
+readonly HADOOP_CONF_DIR='/etc/hadoop/conf'
+readonly HIVE_CONF_DIR='/etc/hive/conf'
+readonly REGION=$(/usr/share/google/get_metadata_value attributes/dataproc-region)
+readonly DEFAULT_INIT_ACTIONS_REPO="gs://dataproc-initialization-actions"
+
+##user supplied location for file to ingest
+readonly INIT_ACTIONS_REPO="$(/usr/share/google/get_metadata_value attributes/init-actions-repo || echo ${DEFAULT_INIT_ACTIONS_REPO})"
+##directory files ingestied will reside
+readonly INIT_ACTIONS_DIR=$(mktemp -d -t dataproc-init-actions-XXXX)
+
+function pre_flight_checks(){
+
+    ##check for bad configurations
+    if [[ "${NUM_LLAP_NODES}" -ge "${WORKER_NODE_COUNT}" ]]; then
+        echo "LLAP node count equals total worker count. There are no nodes to support Tez AM's. Please reduce LLAP instance count and re-deploy." && exit 1
+    fi
+
+}
+
+##add xml doc tool for editing hadoop configuration files
+function configure_yarn_site(){
+    echo "configure yarn-site.xml..."
+
+    bdconfig set_property \
+    --configuration_file "/etc/hadoop/conf/yarn-site.xml" \
+    --name "yarn.application.classpath" \
+    --value "\$HADOOP_CONF_DIR,\$HADOOP_COMMON_HOME/*,\$HADOOP_COMMON_HOME/lib/*,\$HADOOP_HDFS_HOME/*,\$HADOOP_HDFS_HOME/lib/*,\$HADOOP_MAPRED_HOME/*,\$HADOOP_MAPRED_HOME/lib/*,\$HADOOP_YARN_HOME/*,\$HADOOP_YARN_HOME/lib/*,/usr/local/share/google/dataproc/lib/*,\$HADOOP_CONF_DIR,/usr/local/share/google/dataproc/lib/*,/usr/lib/hadoop/*,/usr/lib/hadoop-mapreduce/*,/usr/lib/hadoop/lib/*,/usr/lib/hadoop-hdfs/*,/usr/lib/hadoop-hdfs/lib/*,/usr/lib/hadoop-yarn/*,/usr/lib/hadoop-yarn/lib/*,/usr/lib/tez/*,/usr/lib/tez/lib/*" \
+    --clobber
+
+    if [[ "${NODE_MANAGER_MEMORY}" != "${YARN_MAX_CONTAINER_MEMORY}" ]]; then
+       echo "not configured properly..."
+    fi
+}
+
+function download_init_actions() {
+    # Download initialization actions locally. Check if metadata is supplied
+    echo "downalod init actions supplied as metadata..."
+    mkdir "${INIT_ACTIONS_DIR}"/hive-llap
+    gsutil -m rsync -r "${INIT_ACTIONS_REPO}/hive-llap/" "${INIT_ACTIONS_DIR}/hive-llap/"
+    find "${INIT_ACTIONS_DIR}" -name '*.sh' -exec chmod +x {} \;
+}
+
+###add configurations to hive-site for LLAP
+function configure_hive_site(){
+
+    ##different configuration if HA
+    echo "configure hive-site.xml...."
+    if [[ -n "$ADDITIONAL_MASTER" ]]; then
+        echo "HA deployment .... "
```

Comment:
We can remove some of these echo statements.

Reply:
done